lib/internal/modules/cjs/loader.js

```diff
@@ -78,6 +79,7 @@ const manifest = getOptionValue('--experimental-policy') ?
   require('internal/process/policy').manifest :
   null;
 const { compileFunction } = internalBinding('contextify');
+const { packageInternalResolve } = require('internal/modules/esm/resolve');
```

Comment:
nit: If this is shared, would it make sense to put it into `internal/modules/resolve`? Not sure if it's worth creating a new file for it but it may make it clearer what's ESM/CJS specific when working in the file.

Reply:
The thing is, the ESM resolver functions are still URL-specific, so we go through a `fileURLToPath` conversion process in and out. So still treating these as the "ESM resolver functions" makes sense to me. This also exactly matches the way we call out to the ESM resolver functions from the CJS spec.